react-front-end/src/components/Game.js

```diff
@@ -1,25 +1,100 @@
-import React, { useState, useEffect } from 'react';
-import axios from 'axios';
+import React, { useState, useEffect, useContext } from 'react';
 import "./Game.scss"
 import { useParams } from 'react-router-dom';
-import { grabGameById } from '../helpers/dbHelpers';
+import { getUserId, grabGameById, grabTopReviewsById, grabUserGameLikeFollow } from '../helpers/dbHelpers';
 
-export default function Game(props) {
-  const [game, setGame] = useState("")
+//import context
+import { authContext } from "../providers/AuthProvider";
+
+//components for the GameInformation container
+import Review from "./GamePage/Review"
+import GameDescription from './GamePage/GameDescription';
+import UserButtons from './GamePage/UserButtons';
+
+export default function Game() {
+
+  //import context
+  const { username, token } = useContext(authContext)
+
+  const [game, setGame] = useState({
+    gameData : {},
+    reviewsData: [],
+    userGameData : {}
+
+  })
   const { id } = useParams();
+  const [reviewInputMode, setReviewInputMode] = useState("GameDescription")
   // let game = ''
-
+  // state of star rating
+  
   useEffect(() => {
-    grabGameById(id)
-    .then((result) => {
-      setGame(result[0])
-    });
-  }, []);
+    if (username) {
+      getUserId(username).then((data)=>{
+        const userId = data[0].id
+        Promise.all([
+          grabGameById(id),
+          grabTopReviewsById(id),
+          grabUserGameLikeFollow(userId,id),
+        ]).then((all)=>{
+          const gameData = all[0][0]
+          const reviewsData = all[1]
+          const {liked, played, user_id} = all[2][0] ? all[2][0] : {liked:false, played:false, user_id : userId}
+  
+          setGame({gameData, reviewsData, userGameData : {liked, played, user_id}})
+        }).catch(err => {
+          console.log("ERROR", err.message)// .json({ error: err.message });
+        });
+      }).catch(err => {
+        console.log("ERROR", err.message)// .json({ error: err.message });
+      });
+    } else if (!username) {
+      Promise.all([
+        grabGameById(id),
+        grabTopReviewsById(id),
+      ]).then((all)=>{
+        const gameData = all[0][0]
+        const reviewsData = all[1]
+
+        setGame({gameData, reviewsData})
+      }).catch(err => {
+        console.log("ERROR", err.message)// .json({ error: err.message });
+      });
+    }
+
+  }, [reviewInputMode]);
 
```

Comment:
Everything up here looks good to me.  As long as the set state accesses the game from the db and our data's the same.